### PR TITLE
Add support for ExtInfo Packet

### PR DESCRIPTION
**Pull request by NotAFile**
as documented in https://github.com/piqueserver/aosprotocol/pull/22

### CODE REVIEW

File: Sources/Client/NetClient.h [hunk 1]
```diff
@@ -69,6 +76,10 @@ namespace spades {
 			std::shared_ptr<GameProperties> properties;
 
 			int protocolVersion;
+			std::unordered_map<uint8_t, uint8_t> extensions;
+			std::unordered_map<uint8_t, uint8_t> implementedExtensions {
```
Comment (yvt) [inline Sources/Client/NetClient.h]: Where is this `implementedExtensions` accessed from?

File: Sources/Client/NetClient.h [hunk 2]
```diff
@@ -69,6 +76,10 @@ namespace spades {
 			std::shared_ptr<GameProperties> properties;
 
 			int protocolVersion;
+			std::unordered_map<uint8_t, uint8_t> extensions;
```
Comment (yvt) [inline Sources/Client/NetClient.h]: Add a short description.

File: Sources/Client/NetClient.cpp [hunk 3]
```diff
@@ -714,6 +716,25 @@ namespace spades {
 			}
 		}
 
+		void NetClient::HandleExtensionPacket(spades::client::NetPacketReader& reader) {
+			std::unordered_map<uint8_t, uint8_t> implementedExtensions ( {{192, 1}, {120, 1}} );
```
Comment (yvt) [inline Sources/Client/NetClient.cpp]: Add `static const` because it's constant.

Comment (NotAFile) [reply Sources/Client/NetClient.cpp]: Sorry, it seems I messed up the patch once again. This wasn't supposed to be here.

**Comment by NotAFile** [comment]
Should be good to go now, once the tests run through.

**Pull request merged**